runtime/builtin_types_test.go

```diff
@@ -23,6 +23,33 @@ import (
 	"testing"
 )
 
+func TestBuiltinDelAttr(t *testing.T) {
+	fun := wrapFuncForTest(func(f *Frame, args ...*Object) (*Object, *BaseException) {
+		fooObject := newTestClass("Foo", []*Type{ObjectType}, toDictUnsafe(args[0])).ToObject()
```

Comment:
Rather than creating a test class for each test case, I would create a single test class and create objects of that type that you create inside TestBuiltinDelAttr. Classes are relatively expensive to create and I think we mostly want to test normal object delattr rather than delattr on types.

I just realized you did something similar in SetAttr (sorry for not noticing) but I think we should change that one too. That doesn't need to happen in this PR though.